instrumentation/spring-amqp/src/main/java/brave/spring/amqp/TracingRabbitListenerAdvice.java

```diff
@@ -0,0 +1,47 @@
+package brave.spring.amqp;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext.Extractor;
+import brave.propagation.TraceContextOrSamplingFlags;
+
+public class TracingRabbitListenerAdvice implements MethodInterceptor {

```

Comment:
Maybe add some javadoc on why AOP (I know it was discussed before, but the code should know about it)